Generic string method deprecation

cleidigh opened the pull request:
Firefox removed String generics In 68

jobisoft commented:
I am getting a fatel error with TB68 (release channel), if I use String.trim() or any of the other generics you listed. What version of TB are you using? Can you change the wording, if this is fatal instead of non-fatal?

cleidigh commented:
@jobisoft 
So much for repeating Mozilla some documentation.  I think it must've changed from morning to fatal.
I will double check and adjust the text accordingly.

cleidigh commented:
@jobisoft 
it seems they went from a warning to an exception.  I adjusted the text and added a note about existing methods not changing if they were already using that paradigm.
